ee/codegen/src/generators/nodes/search-node.ts

```diff
@@ -259,11 +283,10 @@ export class SearchNode extends BaseNode<
       VellumValueLogicalExpressionSerializer.parse(metadataFilters);
 
     if (!parsedData.ok) {
-      throw new NodeAttributeGenerationError(
-        `Failed to parse metadata filter JSON: ${JSON.stringify(
-          parsedData.errors
-        )}`
+      const errorMessage = this.formatMetadataFilterParsingError(
+        parsedData.errors
       );
```

Comment:
We don't need to format. All we want to do is to add the error to the `workflowContext.addError()` instead and then return `undefined` here